### PR TITLE
feat(api): homogenize trigger event use cases and fix tech debt

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce?

Homogenise the Trigger Event module use cases and fixes some tech debt related.
Also moves MessageMatcher to its own location as being an use case and not being used by TriggerEvent usecase.

### Why was this change needed?

Tech debt.

### CODE REVIEW

File: apps/api/src/app/events/usecases/index.ts [hunk 1]
```diff
@@ -56,7 +54,6 @@ export const USE_CASES = [
   SendMessageDelay,
   SendTestEmail,
   MapTriggerRecipients,
-  MapTriggerRecipientsCommand,
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/index.ts:59]: This was a mistake I introduced time ago.

File: apps/api/src/app/events/usecases/send-message/send-test-email.usecase.ts [hunk 2]
```diff
@@ -21,7 +23,7 @@ export class SendTestEmail {
     private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations
   ) {}
 
-  public async execute(command: TestSendMessageCommand) {
+  public async execute(command: SendTestEmailCommand) {
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-test-email.usecase.ts:26]: We had a mixing naming style for this use case and command. Homogenised.

File: libs/shared/src/types/builder/filter.types.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import type { BuilderFieldOperator } from './builder.types';
 
+export enum TimeOperatorEnum {
```
Comment (p-fernandez) [inline libs/shared/src/types/builder/filter.types.ts:3]: Added for tighter validation.

File: apps/api/src/app/events/usecases/message-matcher/message-matcher.usecase.ts [hunk 4]
```diff
@@ -458,7 +460,7 @@ async function findAsync<T>(array: T[], predicate: (t: T) => Promise<boolean>):
 }
 
 async function filterAsync<T>(arr: T[], callback: (item: T) => Promise<boolean>): Promise<T[]> {
-  const fail = Symbol();
+  const fail = Symbol('Filter Async failure');
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/message-matcher/message-matcher.usecase.ts:463]: DeepSource flagged this.

File: apps/api/src/app/events/usecases/send-message/send-test-email.command.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+export class SendTestEmailCommand extends EnvironmentWithUserCommand {
+  @IsDefined()
+  contentType: 'customHtml' | 'editor';
+  @IsDefined()
+  payload: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  @IsDefined()
+  @IsString()
+  subject: string;
+  preheader?: string;
+  senderName?: string;
```
Comment (djabarovgeorge) [inline apps/api/src/app/events/usecases/send-message/send-test-email.command.ts:15]: Should we add IsOptional() here?

File: libs/shared/src/types/builder/filter.types.ts (hunk 3 above)
Comment (djabarovgeorge) [reply libs/shared/src/types/builder/filter.types.ts:3]: Loved it 👍🏻

**Review by djabarovgeorge — APPROVED** [review]
Looks good to me, added small comment there

**Pull request merged**